### PR TITLE
SLO validation with payload & chaos

### PR DESCRIPTION
**Pull request by sriumcp**
1. Create SLO validation tutorial with build-in metrics & payload for POST APIs. Partly addresses #1000 
2. Cleanup chaos tutorial. Partly addresses #935 
3. Demonstrate the use of `iter8ctl debug`. Partly addresses #908 
4. Cleanup `mkdocs.yml` bugs
5. Remove links for community meeting agenda and notes, for now (many of the entries are empty/sparse ATM).

### CODE REVIEW

File: mkdocs/docs/contributing/overview.md [hunk 1]
```diff
@@ -44,9 +44,10 @@ Not everything happens through a GitHub pull request. Please come to our
 invite to attend. In fact, we want you to join us, even if you don’t have
 anything you feel like you want to contribute. Just being there is enough!
 - Iter8 community meetings are on the 1st and 3rd Wednesdays of each month from 11 AM - 12 PM EST/EDT. You can find the video
-conference link [here](https://meet.google.com/ocg-bzoa-gqe). You can find the
-agenda [here](https://drive.google.com/drive/folders/1khHMD7JKt-kNAkbLgiIfwDbJYAhHMkyg?usp=sharing) as well as the meeting notes [here](https://drive.google.com/drive/folders/1rVrheoO-nfUI7JdPEgCqM39rl59KrPBk?usp=sharing).
-- If you would like to present a demo or discuss a topic, please list it on the agenda along with a time estimate. Community meetings are recorded and publicly available on our [YouTube channel](https://www.youtube.com/channel/UCVybpnQAhr1o-QRPHBNdUgg).
+conference link [here](https://meet.google.com/ocg-bzoa-gqe). 
+- If you want to discuss an Iter8-related topic, please feel free to do so in any Iter8 community meeting.
```
Comment (Alan-Cha) [inline mkdocs/docs/contributing/overview.md]: Can we link the community meeting?

```
[Iter8 community meeting](https://iter8.tools/0.7/contributing/overview/#come-to-iter8-community-meetings)
```

Comment (Alan-Cha) [reply mkdocs/docs/contributing/overview.md]: Additionally, I just noticed that the "Meet" button at the bottom of the front page goes to [here](https://iter8.tools/0.7/getting-started/help/). Is it more appropriate to point people [here](https://iter8.tools/0.7/contributing/overview/#come-to-iter8-community-meetings)?

And this goes for other similar links in the documentation.

File: mkdocs/docs/contributing/overview.md [hunk 2]
```diff
@@ -44,9 +44,10 @@ Not everything happens through a GitHub pull request. Please come to our
 invite to attend. In fact, we want you to join us, even if you don’t have
 anything you feel like you want to contribute. Just being there is enough!
 - Iter8 community meetings are on the 1st and 3rd Wednesdays of each month from 11 AM - 12 PM EST/EDT. You can find the video
-conference link [here](https://meet.google.com/ocg-bzoa-gqe). You can find the
-agenda [here](https://drive.google.com/drive/folders/1khHMD7JKt-kNAkbLgiIfwDbJYAhHMkyg?usp=sharing) as well as the meeting notes [here](https://drive.google.com/drive/folders/1rVrheoO-nfUI7JdPEgCqM39rl59KrPBk?usp=sharing).
-- If you would like to present a demo or discuss a topic, please list it on the agenda along with a time estimate. Community meetings are recorded and publicly available on our [YouTube channel](https://www.youtube.com/channel/UCVybpnQAhr1o-QRPHBNdUgg).
+conference link [here](https://meet.google.com/ocg-bzoa-gqe). 
+- If you want to discuss an Iter8-related topic, please feel free to do so in any Iter8 community meeting. 
+- If you want to present a demo, please drop us a note in the Iter8 slack workspace (`#general` channel).
```
Comment (Alan-Cha) [inline mkdocs/docs/contributing/overview.md]: Capitalize "Slack" and add a link to the workspace.

```
[Iter8 Slack workspace](https://join.slack.com/t/iter8-tools/shared_invite/zt-awl2se8i-L0pZCpuHntpPejxzLicbmw)
```

File: mkdocs/docs/getting-started/first-experiment.md [hunk 3]
```diff
@@ -67,25 +67,77 @@ helm upgrade -n default my-exp $ITER8/samples/first-exp \
...
+      duration:
+        intervalSeconds: 1
+        iterationsPerLoop: 1
+      versionInfo:
+        # information about app versions used in this experiment
+        baseline:
+          name: my-app
+    ```    
+
+    The [`metrics/collect` task](../reference/tasks/metrics-collect.md) highlighted above is responsible for generating and sending HTTP requests to the app's API end-point, receiving responses, and creating latency and error-related metrics. The [`objectives`](../../concepts/buildingblocks/#objectives-slos) stanza describes the SLOs that the app needs to satisfy in order to be declared a winner.
```
Comment (Alan-Cha) [inline mkdocs/docs/getting-started/first-experiment.md]: No hyphen in "endpoint".

File: mkdocs/docs/getting-started/first-experiment.md [hunk 4]
```diff
@@ -147,9 +212,13 @@ kubectl delete -n default -f $ITER8/samples/deployments/app/deploy.yaml
 
 **Next Steps**
 
-!!! tip "Use with your app"
-    1. Run the above experiment with your app by setting the `URL` value in the Helm command to the URL of your app. 
+!!! tip "Use in your environment"
```
Comment (Alan-Cha) [inline mkdocs/docs/getting-started/first-experiment.md]: I don't like title of the tip. I think it could be more description. Maybe "Try it in your environment"?

File: mkdocs/docs/getting-started/first-experiment.md [hunk 5]
```diff
@@ -147,9 +212,13 @@ kubectl delete -n default -f $ITER8/samples/deployments/app/deploy.yaml
...
+    1. Run the above experiment with your app by setting the `URL` value in the Helm command to the URL of your app. You can run this experiment in any Kubernetes environment such as a dev, test, staging, or production cluster.
```
Comment (Alan-Cha) [inline mkdocs/docs/getting-started/first-experiment.md:217]: Nitpicking but I don't like the value of `URL`.

"By setting the `URL` in the Helm command..."

Comment (Alan-Cha) [reply mkdocs/docs/getting-started/first-experiment.md:217]: Is running things in a production cluster something we want to explicitly suggest?

File: mkdocs/docs/getting-started/help.md [hunk 6]
```diff
@@ -14,7 +14,6 @@ hide:
     - Everyone is welcome to join our community meetings.
     - Our community meetings are on the 1st and 3rd Wednesdays of each month from 11 AM to 12 PM EST/EDT.
         * [Video conference link for community meetings](https://meet.google.com/ocg-bzoa-gqe)
-        * [Agenda](https://drive.google.com/drive/folders/1khHMD7JKt-kNAkbLgiIfwDbJYAhHMkyg?usp=sharing)
-        * [Meeting notes](https://drive.google.com/drive/folders/1rVrheoO-nfUI7JdPEgCqM39rl59KrPBk?usp=sharing)
-    - If you would like to present a demo or discuss a topic, please list it on the agenda along with a time estimate.
+    - If you want to discuss an Iter8-related topic, please feel free to do so in any Iter8 community meeting.
```
Comment (Alan-Cha) [inline mkdocs/docs/getting-started/help.md]: Links similar to what I suggested earlier.

File: mkdocs/docs/tutorials/deployments/slo-validation-chaos.md [hunk 7]
```diff
@@ -1,14 +1,14 @@
...
 
-    **Solution:** You will launch a Kubernetes application along with a composite experiment, consisting of [Litmus Chaos](https://litmuschaos.io/) experiment resource and an Iter8 experiment resource. The chaos experiment will delete pods of the application periodically, while the Iter8 experiment will send requests to the application and verify if it is able to satisfy SLOs.
+    **Solution:** You will launch a Kubernetes application along with a Helm chart consisting of a [Litmus Chaos](https://litmuschaos.io/) experiment, and an Iter8 experiment. The chaos experiment will delete pods of the application periodically, while the Iter8 experiment will send requests to the app and verify if it is able to satisfy SLOs.
 
     ![Chaos with SLO Validation](../../images/chaos-slo-validation.png)
 
 ??? warning "Setup Kubernetes cluster and local environment"
-    0. If you completed the [Iter8 getting-started tutorial](../../getting-started/first-experiment.md) (highly recommended), you may skip to step number 6.
+    0. Complete the [getting started tutorial](../../getting-started/first-experiment.md) (highly recommended), and skip to step 7 below.
```
Comment (Alan-Cha) [inline mkdocs/docs/tutorials/deployments/slo-validation-chaos.md]: I don't see zero-based indexing anywhere else in the documentation...

File: mkdocs/docs/tutorials/deployments/slo-validation-chaos.md [hunk 8]
```diff
@@ -50,12 +50,14 @@ helm upgrade -n default my-exp $ITER8/samples/chaos \
   --install
 ```
 
-## 3. Observe Experiment
-View the chaos resources (chaos engine resource, pod deletion experiment, and RBAC resources), and Iter8 experiment resource.
+The above command creates a [Litmus chaos experiment](https://litmuschaos.io/) and an [Iter8 experiment](../../concepts/whatisiter8.md#what-is-an-iter8-experiment). The former injects chaos into your enviornment by periodically killing pods of your app. The latter generates HTTP requests, collects latency and error rate metrics for the app, and verifies if the app satisfies mean latency (50 msec), error rate (0.0), 95th percentile tail latency SLO (100 msec) SLOs, even in the midst of chaos.
```
Comment (Alan-Cha) [inline mkdocs/docs/tutorials/deployments/slo-validation-chaos.md]: "into your enviornment by periodically"

"environment" misspelling.

Comment (Alan-Cha) [reply mkdocs/docs/tutorials/deployments/slo-validation-chaos.md]: "95th percentile tail latency SLO (100 msec) SLOs" 

SLO repeated

File: mkdocs/docs/tutorials/deployments/slo-validation-chaos.md [hunk 9]
```diff
@@ -105,8 +110,8 @@ kubectl delete -n default -f $ITER8/samples/deployments/app/deploy.yaml
 **Next Steps**
 
 ???+ tip "Use with your own app, and try other types of Chaos"
-    1. You can easily replace the `hello` app used in this tutorial with your own application. a) Modify [Step 1](#1-create-app) to use your service and deployment. b) Modify [Step 2](#2-create-composite-experiment) by supplying the correct namespace and label for your app, and also the correct URL where the app receives requests.
+    1. You can easily replace the `hello` app used in this tutorial with your own application. a) Modify [Step 1](#1-create-app) to use your service and deployment. b) Modify [Step 2](#2-create-joint-experiment) by supplying the correct namespace and label for your app, and also the correct URL where the app receives requests.
```
Comment (Alan-Cha) [inline mkdocs/docs/tutorials/deployments/slo-validation-chaos.md]: Sublist?

File: mkdocs/docs/getting-started/first-experiment.md [hunk 10]
```diff
@@ -4,16 +4,16 @@ template: main.html
...
     
-??? warning "Setup K8s cluster and local environment"
-    1. Setup [K8s cluster](setup-for-tutorials.md#local-kubernetes-cluster)
-    2. [Install Iter8 in K8s cluster](install.md)
+??? warning "Setup Kubernetes cluster and local environment"
+    1. Setup [Kubernetes cluster](setup-for-tutorials.md#local-kubernetes-cluster)
+    2. [Install Iter8 in Kubernetes cluster](install.md)
     3. Get [Helm 3.4+](https://helm.sh/docs/intro/install/).
     4. Get [`iter8ctl`](install.md#get-iter8ctl)
     5. Fork the [Iter8 GitHub repo](https://github.com/iter8-tools/iter8). Clone your fork, and set the ITER8 environment variable as follows.
```
Comment (Alan-Cha) [inline mkdocs/docs/getting-started/first-experiment.md]: `set the ITER8 environment variable`

I think `ITER8` should have code syntax. I think there may be other instances of this in the docs.

File: mkdocs/docs/tutorials/deployments/slo-validation-chaos.md [hunk 11]
```diff
@@ -1,14 +1,14 @@
...
+    **Solution:** You will launch a Kubernetes application along with a Helm chart consisting of a [Litmus Chaos](https://litmuschaos.io/) experiment, and an Iter8 experiment. The chaos experiment will delete pods of the application periodically, while the Iter8 experiment will send requests to the app and verify if it is able to satisfy SLOs.
 
     ![Chaos with SLO Validation](../../images/chaos-slo-validation.png)
 
 ??? warning "Setup Kubernetes cluster and local environment"
-    0. If you completed the [Iter8 getting-started tutorial](../../getting-started/first-experiment.md) (highly recommended), you may skip to step number 6.
+    0. Complete the [getting started tutorial](../../getting-started/first-experiment.md) (highly recommended), and skip to step 7 below.
     1. Setup [Kubernetes cluster](../../getting-started/setup-for-tutorials.md#local-kubernetes-cluster)
     2. [Install Iter8 in Kubernetes cluster](../../getting-started/install.md)
```
Comment (Alan-Cha) [inline mkdocs/docs/tutorials/deployments/slo-validation-chaos.md]: "[Install Iter8 in Kubernetes cluster](../../getting-started/install.md)" is a hyperlink whereas only "Litmus" is a hyperlink in "Install [Litmus](https://litmuschaos.io/) in Kubernetes cluster".

IMO, I think the latter fits better with the rest of the list. This needs to be changed in multiple places.

File: mkdocs/docs/tutorials/deployments/slo-validation-chaos.md [hunk 12]
```diff
@@ -105,8 +110,8 @@ kubectl delete -n default -f $ITER8/samples/deployments/app/deploy.yaml
...
+    2. Litmus makes it possible to inject [over 51 types of Chaos](https://hub.litmuschaos.io/). Modify the Helm chart to use any of these other types of chaos experiments.
 
-    3. Iter8 makes it possible to [promote the winning version](../../concepts/buildingblocks.md#version-promotion) in a number of different ways. For example, you may have a stable version running in production, a candidate version deployed in a staging environment, perform this experiment, ensure that the candidate is successful, and promote it as the latest stable version in a GitOps-y manner as described [here](../deployments/slo-validation-gitops.md).
+    3. Iter8 makes it possible to [promote the winning version](../../concepts/buildingblocks.md#version-promotion) in a number of different ways. For example, you may have a stable version running in production, have a candidate version deployed in a staging environment, perform this experiment, ensure that the candidate is successful, and promote it as the latest stable version in a GitOps-y manner as described [here](../deployments/slo-validation-gitops.md).
```
Comment (Alan-Cha) [inline mkdocs/docs/tutorials/deployments/slo-validation-chaos.md]: Can we reword this to "have a stable version running in production and a candidate version deployed in a staging environment"? I think it has a better flow this way.

File: samples/deployments/payload/values.yaml [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+time: 5s
+
+# Queries per second used by Iter8 for sending requests
+QPS: 8.0
+
+# Upper limit on the mean latency. Values beyond this limit are not acceptable
+limitMeanLatency: 500.0 # 500 msec latency on an average is ok. Not more.
+
+# Upper limit on the error rate. Values beyond this limit are not acceptable
+limitErrorRate: 0.00 # Zero %. Errors are not acceptable.
```
Comment (Alan-Cha) [inline samples/deployments/payload/values.yaml]: I think 0% might read better.

File: samples/deployments/payload/values.yaml [hunk 14]
```diff
@@ -0,0 +1,19 @@
+# Default values for Iter8 SLO validation experiment
+
+# URL of the application for which this SLO validation experiment is performed
+# URL: https://example.com
+
+# The duration for which Iter8 will send requests to the app
```
Comment (Alan-Cha) [inline samples/deployments/payload/values.yaml]: Remove the "the"

File: mkdocs/docs/contributing/overview.md (hunk 1 above)
Comment (sriumcp) [reply mkdocs/docs/contributing/overview.md]: Latest changes fix this.

File: mkdocs/docs/getting-started/first-experiment.md (hunk 5 above)
Comment (sriumcp) [reply mkdocs/docs/getting-started/first-experiment.md:217]: Production suggestion is good to include, I think. At the minimum, for the sake of completeness.

Comment (sriumcp) [reply mkdocs/docs/getting-started/first-experiment.md:217]: Will follow up with another PR to fix the URL value. I don't like the value of the URL either :)

File: mkdocs/docs/getting-started/help.md (hunk 6 above)
Comment (sriumcp) [reply mkdocs/docs/getting-started/help.md]: Fixed.

File: mkdocs/docs/tutorials/deployments/slo-validation-chaos.md (hunk 8 above)
Comment (sriumcp) [reply mkdocs/docs/tutorials/deployments/slo-validation-chaos.md]: Fixed.

File: mkdocs/docs/getting-started/first-experiment.md (hunk 10 above)
Comment (sriumcp) [reply mkdocs/docs/getting-started/first-experiment.md]: Fixed.

File: mkdocs/docs/tutorials/deployments/slo-validation-chaos.md (hunk 11 above)
Comment (sriumcp) [reply mkdocs/docs/tutorials/deployments/slo-validation-chaos.md]: Minor difference ... the former is actually a link for installation instructions for Iter8 (not the Iter8 project). The latter is a link for the Litmus project, and not for its install instructions, which are explicitly provided in the next line. Hence, the difference.

File: mkdocs/docs/tutorials/deployments/slo-validation-chaos.md (hunk 12 above)
Comment (sriumcp) [reply mkdocs/docs/tutorials/deployments/slo-validation-chaos.md]: Fixed.

File: samples/deployments/payload/values.yaml (hunk 14 above)
Comment (sriumcp) [reply samples/deployments/payload/values.yaml]: Done.

File: mkdocs/docs/reference/tasks/metrics-collect.md [hunk 15]
```diff
@@ -32,17 +23,21 @@ start:
 ## Inputs
 | Field name | Field type | Description | Required |
 | ----- | ---- | ----------- | -------- |
-| time | string | Duration of the `metrics/collect` task run. Specified in the [Go duration string format](https://golang.org/pkg/time/#ParseDuration). Default value is `5s`. | No |
-| payloadURL | string | URL of JSON-encoded data. If this field is specified, the metrics collector will send HTTP POST requests to versions, and the POST requests will contain this JSON data as payload. | No |
-| versions | [][Version](#version) | A non-empty list of versions. | Yes |
+| numQueries | int | number of requests sent to each version. Default value is 100. | No |
```
Comment (kalantar) [inline mkdocs/docs/reference/tasks/metrics-collect.md]: Is this number of requests *to be* sent to each version?

Comment (kalantar) [reply mkdocs/docs/reference/tasks/metrics-collect.md]: Since this has precedence over time, perhaps that should be stated explicitly.

File: mkdocs/docs/tutorials/deployments/slo-validation-chaos.md [hunk 16]
```diff
@@ -77,23 +79,26 @@ Due to chaos injection, and the fact that the number of replicas of the app in t
 iter8ctl assert -c winnerFound
```
Comment (kalantar) [inline mkdocs/docs/tutorials/deployments/slo-validation-chaos.md]: why do we not use the `-c completed` assertion as well? I noticed it in the first experiment and would expect it to be the standard query.

File: mkdocs/docs/tutorials/deployments/slo-validation-chaos.md [hunk 17]
```diff
@@ -77,23 +79,26 @@ Due to chaos injection, and the fact that the number of replicas of the app in t
...
 kubectl scale --replicas=2 -n default -f $ITER8/samples/deployments/app/deploy.yaml
 ```
 
+The scaled app is now more resilient. Performing the same experiment as above will now result in SLOs being satisfied and a winner being found.
+
 Retry steps 2 and 3 above. You should now find that SLOs are satisfied and a winner is found at the end of the experiment.
 
+```shell
+# this assertion is expected to succeed
+iter8ctl assert -c winnerFound
```
Comment (kalantar) [inline mkdocs/docs/tutorials/deployments/slo-validation-chaos.md]: same question

File: mkdocs/docs/getting-started/first-experiment.md (hunk 4 above)
Comment (sriumcp) [reply mkdocs/docs/getting-started/first-experiment.md]: Fixed.

File: mkdocs/docs/tutorials/deployments/slo-validation-chaos.md (hunk 9 above)
Comment (sriumcp) [reply mkdocs/docs/tutorials/deployments/slo-validation-chaos.md]: Done.

File: samples/deployments/payload/values.yaml (hunk 13 above)
Comment (sriumcp) [reply samples/deployments/payload/values.yaml]: Done.

File: mkdocs/docs/reference/tasks/metrics-collect.md (hunk 15 above)
Comment (sriumcp) [reply mkdocs/docs/reference/tasks/metrics-collect.md]: Fixed `to be sent`. Also explicitly specified number has precedence over time in the time field.

File: mkdocs/docs/tutorials/deployments/slo-validation-chaos.md (hunk 16 above)
Comment (sriumcp) [reply mkdocs/docs/tutorials/deployments/slo-validation-chaos.md]: Fixed.

File: mkdocs/docs/tutorials/deployments/slo-validation-chaos.md (hunk 17 above)
Comment (sriumcp) [reply mkdocs/docs/tutorials/deployments/slo-validation-chaos.md]: Fixed.

**Review by Alan-Cha — DISMISSED** [review]

**Review by kalantar — APPROVED** [review]
lgtm
